src/main/java/hudson/plugins/parameterizedtrigger/BuildTriggerConfig.java

```diff
@@ -403,15 +405,35 @@ protected Cause createUpstreamCause(AbstractBuild<?, ?> build) {
         return new UpstreamCause((Run) build);
     }
 
-    protected Future schedule(AbstractBuild<?, ?> build, AbstractProject project, int quietPeriod, List<Action> list) throws InterruptedException, IOException {
+    protected Future schedule(AbstractBuild<?, ?> build, final Job project, int quietPeriod, List<Action> list) throws InterruptedException, IOException {
         Cause cause = createUpstreamCause(build);
-        return project.scheduleBuild2(quietPeriod,
-                cause,
-                list.toArray(new Action[list.size()]));
+        List<Action> queueActions = new ArrayList<Action>(list);
+        if (cause != null) {
+            queueActions.add(new CauseAction(cause));
+        }
+
+        // Includes both traditional projects via AbstractProject and Workflow Job
+        if (project instanceof ParameterizedJobMixIn.ParameterizedJob) {
+            final ParameterizedJobMixIn<?, ?> parameterizedJobMixIn = new ParameterizedJobMixIn() {
+                @Override
+                protected Job<?, ?> asJob() {
+                    return project;
+                }
+            };
+            return parameterizedJobMixIn.scheduleBuild2(quietPeriod, queueActions.toArray(new Action[queueActions.size()]));
```

Comment:
FYI https://github.com/jenkinsci/jenkins/pull/1771 would be nicer here. I would recommend leaving in a `TODO` comment to that effect.